pkg/agent/plugin/svidstore/aws/aws.go

```diff
@@ -0,0 +1,289 @@
+}
+
+func newPlugin(newClient func(ctx context.Context, secretAccessKey, accessKeyID, region string) (SecretsManagerClient, error)) *SecretsManagerPlugin {
+	p := &SecretsManagerPlugin{}
+	p.hooks.newClient = newClient
+	p.hooks.getenv = os.Getenv
+
+	return p
+}
+
+type secretOptions struct {
+	name     string
+	arn      string
+	kmsKeyID string
+}
+
+func optionsFromSecretData(selectorData []string) (*secretOptions, error) {
+	data := svidstore.ParseMetadata(selectorData)
+
+	opt := &secretOptions{
+		name:     data["secretname"],
+		arn:      data["arn"],
+		kmsKeyID: data["kmskeyid"],
+	}
+
+	if opt.name == "" && opt.arn == "" {
+		return nil, status.Error(codes.InvalidArgument, "secret name or ARN are required")
+	}
+
+	return opt, nil
+}
+
+type Configuration struct {
+	AccessKeyID     string `hcl:"access_key_id" json:"access_key_id"`
+	SecretAccessKey string `hcl:"secret_access_key" json:"secret_access_key"`
+	Region          string `hcl:"region" json:"region"`
+}
+
+type SecretsManagerPlugin struct {
+	svidstorev1.UnsafeSVIDStoreServer
+	configv1.UnsafeConfigServer
+
+	log      hclog.Logger
+	smClient SecretsManagerClient
+	mtx      sync.RWMutex
+
+	hooks struct {
+		newClient func(ctx context.Context, secretAccessKey, accessKeyID, region string) (SecretsManagerClient, error)
+		getenv    func(string) string
+	}
+}
+
+func (p *SecretsManagerPlugin) SetLogger(log hclog.Logger) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	p.log = log
+}
+
+// Configure configures the SecretsManagerPlugin.
+func (p *SecretsManagerPlugin) Configure(ctx context.Context, req *configv1.ConfigureRequest) (*configv1.ConfigureResponse, error) {
+	// Parse HCL config payload into config struct
+	config := &Configuration{}
+	if err := hcl.Decode(config, req.HclConfiguration); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "unable to decode configuration: %v", err)
+	}
+
+	if config.AccessKeyID == "" {
+		config.AccessKeyID = p.hooks.getenv("AWS_ACCESS_KEY_ID")
+	}
+
+	if config.SecretAccessKey == "" {
+		config.SecretAccessKey = p.hooks.getenv("AWS_SECRET_ACCESS_KEY")
+	}
+
+	if config.Region == "" {
+		return nil, status.Error(codes.InvalidArgument, "region is required")
+	}
+
+	smClient, err := p.hooks.newClient(ctx, config.SecretAccessKey, config.AccessKeyID, config.Region)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to create secrets manager client: %v", err)
+	}
+
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	p.smClient = smClient
+
+	return &configv1.ConfigureResponse{}, nil
+}
+
+// PutX509SVID puts the specified X509-SVID in the configured AWS Secrets Manager
+func (p *SecretsManagerPlugin) PutX509SVID(ctx context.Context, req *svidstorev1.PutX509SVIDRequest) (*svidstorev1.PutX509SVIDResponse, error) {
+	opt, err := optionsFromSecretData(req.Metadata)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use arn if it is configured if not use name
+	secretID := opt.name
+	if opt.arn != "" {
+		secretID = opt.arn
+	}
```

Comment:
Looks like this block may be moved to a function that returns a secretID based on the options? This is also used in DeleteX509SVID.